### PR TITLE
[GEOS-9485] GeoServer system-status {json,xml} endpoints do not retur…

### PR DESCRIPTION
**Pull request by taba90**
…n values anymore

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [X] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [X] PR for bug fixes and small new features are presented as a single commit
- [X] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [X] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [X] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by taba90** [comment]
Associated JIRA ticket https://osgeo-org.atlassian.net/browse/GEOS-9485

File: src/restconfig/src/test/java/org/geoserver/rest/system/status/MonitorRestTest.java [hunk 1]
```diff
@@ -69,6 +74,41 @@ public void testJsonCallback() throws Exception {
...
+                getAsServletResponse(RestBaseController.ROOT_PATH + "/about/system-status.json");
+        assertEquals(200, response.getStatus());
+        assertEquals("application/json", response.getContentType());
+        JSONArray array =
+                ((JSONObject)
+                                ((JSONObject) JSONSerializer.toJSON(response.getContentAsString()))
+                                        .get("metrics"))
+                        .getJSONArray("metric");
+        assertTrue(array.size() == MetricInfo.values().length);
+        for (int i = 0; i < array.size(); i++) {
```
Comment (nmco) [inline src/restconfig/src/test/java/org/geoserver/rest/system/status/MonitorRestTest.java]: If my memory serves me well, if a metric is not available in a certain environment (for example the machine doesn't provide a network sensor) the the metric will not be collected. Hence, checking that available metrics have a value makes sense, but checking the size probably not :scream:.

**Comment by nmco** [comment]
Well the test failure confirms my suspicion:
```
[ERROR] testXmlValueFieldIsEncoded(org.geoserver.rest.system.status.MonitorRestTest)  Time elapsed: 1.785 s  <<< FAILURE!
java.lang.AssertionError
	at org.geoserver.rest.system.status.MonitorRestTest.testXmlValueFieldIsEncoded(MonitorRestTest.java:106)

[ERROR] testJsonValueFieldIsEncoded(org.geoserver.rest.system.status.MonitorRestTest)  Time elapsed: 0.687 s  <<< FAILURE!
java.lang.AssertionError
	at org.geoserver.rest.system.status.MonitorRestTest.testJsonValueFieldIsEncoded(MonitorRestTest.java:88)
```

File: src/restconfig/src/test/java/org/geoserver/rest/system/status/MonitorRestTest.java (hunk 1 above)
Comment (taba90) [reply src/restconfig/src/test/java/org/geoserver/rest/system/status/MonitorRestTest.java]: Thanks for your review @nmco, this has  been fixed

Comment (nmco) [reply src/restconfig/src/test/java/org/geoserver/rest/system/status/MonitorRestTest.java]: Great, thx!

**Comment by taba90** [comment]
Build errors don't seems related to code changes in this pull request:
[ERROR] Errors: 
[ERROR]   GroupListPageTest>AbstractTabbedListPageTest.initService:34->AbstractSecurityWicketTestSupport.clearServices:184 � IO
[ERROR]   GroupListPageTest>AbstractTabbedListPageTest.initService:35->AbstractSecurityWicketTestSupport.insertValues:157 � IO
[ERROR]   RoleListPageTest>AbstractTabbedListPageTest.initService:35->AbstractSecurityWicketTestSupport.insertValues:157 � IO
[INFO] 
[ERROR] Tests run: 99, Failures: 0, Errors: 3, Skipped: 0

File: src/main/src/main/java/org/geoserver/system/status/MetricValue.java [hunk 2]
```diff
@@ -34,6 +34,8 @@
 
     int priority;
 
+    ValueHolder holder;
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/system/status/MetricValue.java]: This was removed in https://github.com/geoserver/geoserver/pull/3590/files#diff-1e72fa7cb4d8fce2c903e9686f4533d6L41
What's going on? :-D

Comment (taba90) [reply src/main/src/main/java/org/geoserver/system/status/MetricValue.java]: @aaime  the removal of ValueHolder caused metric values to not be included in json and xml output format, so I put them back

**Comment by aaime** [comment]
@taba90 ok squash-merged. Do you need to run backports?

**Pull request merged**